### PR TITLE
Fix race condition in test_mesh_reconnect

### PR DESCRIPTION
**Pull request by Jauler**
### Description
There is a time gap when meshnet is started in `test_mesh_reconnect` and test starts to wait for derp/node events coming from libtelio. If libtelio fires an even in this time gap when meshnet is already started, but test is still not waiting for event - test will attempt to wait for **second** derp/node event which will never come. This PR removes this time gap.

### :ballot_box_with_check: Definition of Done checklist
- [x] Commit history is clean ([requirements](../blob/main/docs/git_commit_messages_requirements.md))
- [x] README.md is updated
- [x] Functionality is covered by unit or integration tests

### CODE REVIEW

File: nat-lab/tests/test_reconnections.py [hunk 1]
```diff
@@ -109,9 +109,9 @@ async def test_mesh_reconnect(
             await ping(beta_connection, alpha.ip_addresses[0], 15)
 
         await client_alpha.simple_start()
-        await client_alpha.set_meshmap(api.get_meshmap(alpha.id))
 
         await asyncio.gather(
+            client_alpha.set_meshmap(api.get_meshmap(alpha.id)),
```
Comment (gytsto) [inline nat-lab/tests/test_reconnections.py:114]: This doesn't really fix the race condition. But on the other hand, starting to wait for event before action, might fix the race condition:

```
        async with run_async_context(
            asyncio.gather(
                client_alpha.wait_for_event_peer(
                    beta.public_key, [telio.State.Connected]
                ),
                client_alpha.wait_for_event_on_any_derp([telio.State.Connected]),
            )
        ) as events:
            await client_alpha.set_meshmap(api.get_meshmap(alpha.id))
            await events
```

**Comment by mathiaspeters** [comment]
Have you verified that this works? I'm asking because I think it's a common pattern to first set the meshmap and afterwards wait for an event, and I might be mistaken but I'm pretty sure when you wait for a specific event (in this case connected), it doesn't just look at events from the moment the `wait_for_*` was called, but rather it checks the events since the previous time `wait_for_*` was called, which in this case would mean that it would see the first event and wouldn't be waiting. I could be wrong though so if you can somewhat reliably reproduce the issue before this fix and can't repro after, I'll approve the PR

**Comment by gytsto** [comment]
> Have you verified that this works? I'm asking because I think it's a common pattern to first set the meshmap and afterwards wait for an event, and I might be mistaken but I'm pretty sure when you wait for a specific event (in this case connected), it doesn't just look at events from the moment the `wait_for_*` was called, but rather it checks the events since the previous time `wait_for_*` was called, which in this case would mean that it would see the first event and wouldn't be waiting. I could be wrong though so if you can somewhat reliably reproduce the issue before this fix and can't repro after, I'll approve the PR

I think you are mixing up state and event. Before uniffi changes, wait_for_event would waiy until new event is generated without looking to previously generated ones. State would return immidiately if it was in that state before even before the action of set_meshmap. But starting to wait for event, before setting meshmap, would actually solve the race condition.

File: nat-lab/tests/test_reconnections.py (hunk 1 above)
Comment (Jauler) [reply nat-lab/tests/test_reconnections.py]: oh nice, thanks for taking a look :pray: 

Fixed.

File: nat-lab/tests/test_reconnections.py [hunk 2]
```diff
@@ -109,17 +110,17 @@ async def test_mesh_reconnect(
...
 
-        await asyncio.gather(
-            client_alpha.wait_for_event_peer(beta.public_key, [telio.State.Connected]),
-            client_alpha.wait_for_event_on_any_derp([telio.State.Connected]),
-        )
-
-        await asyncio.gather(
-            client_alpha.wait_for_state_peer(beta.public_key, [telio.State.Connected]),
-            client_beta.wait_for_state_peer(alpha.public_key, [telio.State.Connected]),
-        )
```
Comment (gytsto) [inline nat-lab/tests/test_reconnections.py:122]: nit: Would still be nice to check if state is reported correctly on both sides after event is generated, but up to you

**Review by gytsto — DISMISSED** [review]
+2, other than one nit

File: nat-lab/tests/test_reconnections.py (hunk 2 above)
Comment (gytsto) [reply nat-lab/tests/test_reconnections.py:122]: didn't see your comment, this is useless :D

**Review by mathiaspeters — APPROVED** [review]

File: nat-lab/tests/test_reconnections.py (hunk 2 above)
Comment (Jauler) [reply nat-lab/tests/test_reconnections.py:122]: generally `beta` is likely will not generate any events. So this is why I thought that asserting on them may be not the wisest thing

Comment (Jauler) [reply nat-lab/tests/test_reconnections.py:122]: but `.wait_for_state_peer` is safe probably, so I will readd it :D (changed my mind)

**Pull request merged**